### PR TITLE
[14470] Improve doxygen documentation related to Locator classes.

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
## Description

This PR improves doxygen documentation related to Locator classes improving thus the API reference.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
* **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [x] Any new/modified methods have been properly documented using Doxygen. 
* **N/A** Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [x] Documentation builds and tests pass locally. 
* **N/A** New feature has been added to the `versions.md` file (if applicable).
- [x] New feature has been documented/Current behavior is correctly described in the documentation. 
Related documentation PR: eProsima/Fast-DDS-docs#367

## Reviewer Checklist
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please check style

File: include/fastdds/rtps/common/LocatorList.hpp [hunk 1]
```diff
@@ -188,23 +230,43 @@ class LocatorList
...
     RTPS_DllAPI void resize(
             size_t num)
     {
         return m_locators.resize(num);
     }
 
+    /**
+     * @brief Add locator to the end.
+     *
+     * @param loc locator to be appended.
```
Comment (MiguelCompany) [inline include/fastdds/rtps/common/LocatorList.hpp:268]: We should document that `loc` will not be added if it is already present

File: include/fastdds/rtps/common/LocatorList.hpp [hunk 2]
```diff
@@ -223,6 +285,11 @@ class LocatorList
         }
     }
 
+    /**
+     * @brief Add several locators to the end.
+     *
+     * @param locList LocatorList with the locators to be appended.
+     */
```
Comment (MiguelCompany) [inline include/fastdds/rtps/common/LocatorList.hpp:292]: Same here

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: include/fastdds/rtps/common/LocatorList.hpp [hunk 3]
```diff
@@ -296,6 +383,15 @@ class LocatorList
     std::vector<Locator> m_locators;
 };
 
+/**
+ * @brief Insertion operator: serialize a locator list
+ *        Each individual locator within the list must follow the serialization format explained in the locator insertion
```
Comment (MiguelCompany) [inline include/fastdds/rtps/common/LocatorList.hpp:388]: Please mention the separator used between locators

File: include/fastdds/rtps/common/LocatorList.hpp [hunk 4]
```diff
@@ -317,6 +413,15 @@ inline std::ostream& operator <<(
     return output;
 }
 
+/**
+ * @brief Extraction operator: deserialize a list of locators
+ *        Each individual locator within the list must follow the deserialization format explained in the locator
```
Comment (MiguelCompany) [inline include/fastdds/rtps/common/LocatorList.hpp:418]: Please mention the separator used between locators

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Review by MiguelCompany — APPROVED** [review]
LGTM. I've triggered a build-only CI just in case.

**Comment by MiguelCompany** [comment]
Documentation job failed due to doxygen comments here:

```
fastdds/api_reference/rtps/common/Locator/operators.rst:None: (locatorN)  Insertion operator: serialize a locator list. The deserialization format is [locator1,locator2,...,locatorN]. Each individual locator within the list must follow the serialization format explained in the locator insertion operator. 
fastdds/api_reference/rtps/common/Locator/operators.rst:None: (locatorN)  Extraction operator: deserialize a list of locators. The serialization format is [locator1,locator2,...,locatorN]. Each individual locator within the list must follow the deserialization format explained in the locator extraction operator. 
```

**Comment by JLBuenoLopez-eProsima** [comment]
Thanks @MiguelCompany! Fixed already in eProsima/Fast-DDS-docs#367

**Pull request merged**